src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.addAlive(this);
+  }
+
+  static alive = [];
+
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+  static removeDead(animal) {
+    Animal.alive = Animal.alive.filter((aliveAnimal) => aliveAnimal !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
```

Comment:
When reducing the health of the `target`, ensure that the health does not go below zero to avoid negative health values.